### PR TITLE
Scalar operator changes for Tensor on NumpyBackend

### PR DESCRIPTION
**Pull request by miriyalavignesh**
This change has only changes for Numpy Backend will follow up with other backend changes.

### CODE REVIEW

File: tensornetwork/backends/base_backend.py [hunk 1]
```diff
@@ -337,3 +337,51 @@ def multiply(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
...
+        tensor2: A tensor.
+      Returns:
+        Tensor
+      """
+      raise NotImplementedError("Backend '{}' has not implemented add.".format(
+          self.name))
+
+  def sub(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
+      """
+      Return the addition of `tensor` with another `tensor`
```
Comment (MichaelMarien) [inline tensornetwork/backends/base_backend.py]: addition -> subtraction?

File: tensornetwork/backends/base_backend.py [hunk 2]
```diff
@@ -337,3 +337,51 @@ def multiply(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
...
+  def scalar_multiply(self, tensor: Tensor, scalar) -> Tensor:
+      """
+      Return the multiplication of `tensor` with a scalar value
+      Args:
+        tensor: A tensor.
+        scalar: A numeric value.
+      Returns:
+        Tensor
+      """
+      raise NotImplementedError("Backend '{}' has not implemented multiply.".format(
```
Comment (MichaelMarien) [inline tensornetwork/backends/base_backend.py]: multiplt-> scalar multiply (we already have a multiply)

File: tensornetwork/backends/base_backend.py [hunk 3]
```diff
@@ -337,3 +337,51 @@ def multiply(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
...
+  def scalar_divide(self, tensor: Tensor, scalar) -> Tensor:
+      """
+      Return the division of `tensor` with a scalar value
+      Args:
+        tensor: A tensor.
+        scalar: A numeric value.
+      Returns:
+        Tensor
+      """
+      raise NotImplementedError("Backend '{}' has not implemented true_divide.".format(
```
Comment (MichaelMarien) [inline tensornetwork/backends/base_backend.py]: true_divide is np name, let's use the backend method name?

**Comment by MichaelMarien** [comment]
We need to sync this one with #358 as similar functionalities are added to backend.

**Comment by miriyalavignesh** [comment]
Synced with @jziub . He will take care of addition and subtraction and I will make changes for the scalar multiply and divide.

Thanks!

**Comment by Thenerdstation** [comment]
Thanks for the PR!

Other projects have taken priority at this time, so I will review this in ~2 weeks.

**Pull request closed without merging**